### PR TITLE
[BL-35] - Add React Component for Profile Picture

### PR DESCRIPTION
**Pull request by LuisRococo**
* Add user menu on navbar
* Add logout functionality
* add margin on general error component btn

### CODE REVIEW

File: src/components/boardPage/GeneralErrorMessage/GeneralErrorMessage.tsx [hunk 1]
```diff
@@ -12,7 +12,10 @@ export const GeneralErrorMessage: React.FC<IGeneralErrorMessage> = ({
       <div className="text-center mt-5">
         <h3>There was an error</h3>
         <p className="m-0">Maybe you selected an invalid board</p>
-        <button onClick={onSelectededBoardIdReset} className="btn btn-primary">
```
Comment (ohenroute) [inline src/components/boardPage/GeneralErrorMessage/GeneralErrorMessage.tsx:15]: make according change from the last PR.

File: src/components/common/Navigation/Navigation.scss [hunk 2]
```diff
@@ -1,3 +1,62 @@
...
+
+    
+    
+}
+
+.dropdown {
+    position: relative;
+    display: inline-block;
+
+    &:hover .dropdown-content {display: block;}
```
Comment (ohenroute) [inline src/components/common/Navigation/Navigation.scss]: doesn't look so fancy or readable.

File: src/components/common/Navigation/Navigation.tsx [hunk 3]
```diff
@@ -3,36 +3,63 @@ import logoImg from "../../../assets/logo.png";
...
+            GreatBoard
+          </div>
+
+          {userData && (
+            <div className="dropdown">
+              <div className="profile">
+                <img src={profileImg} alt="user-logo" />
+                <div>
+                  <p className="text-white m-0">{userData.firstName}</p>
+                  <p className="user-auth-type m-0">{userData.authType}</p>
```
Comment (ohenroute) [inline src/components/common/Navigation/Navigation.tsx]: From now on, de-structure variables before using them. 

`const { firstName, authType } = userData` 

It's getting messy.

File: src/components/common/Navigation/Navigation.tsx [hunk 4]
```diff
@@ -3,36 +3,63 @@ import logoImg from "../../../assets/logo.png";
...
+                </div>
+              </div>
+
+              <div className="dropdown-content">
+                <ProfileMenuBtn
+                  title="Edit Profile"
+                  Icon={AiFillEdit}
+                  onClick={() => {
+                    console.log("Placeholder");
+                  }}
```
Comment (ohenroute) [inline src/components/common/Navigation/Navigation.tsx:57]: Will this be functional?

File: src/components/common/Navigation/Navigation.tsx [hunk 5]
```diff
@@ -3,36 +3,63 @@ import logoImg from "../../../assets/logo.png";
...
+                />
+                <ProfileMenuBtn
+                  title="Logout"
+                  Icon={MdOutlineLogout}
+                  onClick={() => handleLogOutBtnClick()}
```
Comment (ohenroute) [inline src/components/common/Navigation/Navigation.tsx]: why not invoking the function itself `handleLogOutBtnClick`? You are not using the event.

File: src/hooks/useUserState.ts [hunk 6]
```diff
@@ -32,6 +32,7 @@ export const useUserState = () => {
         email: responseData.email,
         firstName: responseData.first_name,
         lastName: responseData.last_name,
+        authType: responseData.authorization_tier,
```
Comment (ohenroute) [inline src/hooks/useUserState.ts]: Destructure this block of code. 

```
    setUser({
      userData: { id, email, firstName: first_name, lastName: last_name .... }
      },
    });

    // alternatively: 
    
    const { id, first_name: firstName, last_name: lastName ... }
```

File: src/components/common/Navigation/Navigation.tsx (hunk 4 above)
Comment (LuisRococo) [reply src/components/common/Navigation/Navigation.tsx:57]: Yes, one of the tasks is to have the option for the user to update his information.

**Comment by LuisRococo** [comment]
Requested changes added.

File: src/hooks/useUserState.ts [hunk 7]
```diff
@@ -25,13 +25,15 @@ export const useUserState = () => {
...
 
     setUser({
       userData: {
-        id: responseData.id,
-        email: responseData.email,
+        id,
+        email,
         firstName: responseData.first_name,
         lastName: responseData.last_name,
+        authType: responseData.authorization_tier,
```
Comment (ohenroute) [inline src/hooks/useUserState.ts]: De-structure all variables, including these with different names. 

```
const {
    id, 
    email, 
    first_name: firstName, 
    last_name: lastName, 
    authorization_tier: authType
} = responseData;```

**Review by ohenroute — CHANGES_REQUESTED** [review]
Just an additional small fix.

**Review by ohenroute — APPROVED** [review]

**Pull request merged**